### PR TITLE
Support `execute if function` and future `execute if` conditions.

### PR DESCRIPTION
**Pull request by Nico314159**

### CODE REVIEW

File: src/jmc/compile/command/condition.py [hunk 1]
```diff
@@ -10,6 +10,8 @@
 from ..datapack import DataPack
 from .utils import find_scoreboard_player_type, PlayerType, is_obj_selector, merge_obj_selector
 from .jmc_function import JMCFunction, FuncType
+from jmc.compile.vanilla_command import CONDITIONS as VANILLA_CONDITIONS
```
Comment (WingedSeal) [inline src/jmc/compile/command/condition.py:13]: Instead of importing it here, initialize header with it

File: src/jmc/compile/header.py [hunk 2]
```diff
@@ -85,6 +87,7 @@ def __clear(obj: "Header"):
         obj.is_enable_macro = True
         obj.namespace_overrides = set()
         obj.commands = set()
+        obj.conditions = set()
```
Comment (WingedSeal) [inline src/jmc/compile/header.py:90]: move `CONDITION` to here instead so you don't have to merge it every token.

File: src/jmc/compile/vanilla_command.py [hunk 3]
```diff
@@ -88,3 +88,15 @@
     "whitelist"
 }
 """All vanilla command (first argument only)"""
+CONDITIONS = {
```
Comment (WingedSeal) [inline src/jmc/compile/vanilla_command.py:91]: Consider giving it a comment and renaming it to `VANILLA_CONDITIONS`

**Review by WingedSeal — CHANGES_REQUESTED** [review]

**Comment by Nico314159** [comment]
Made the changes

File: src/jmc/compile/header.py [hunk 4]
```diff
@@ -85,6 +88,7 @@ def __clear(obj: "Header"):
         obj.is_enable_macro = True
         obj.namespace_overrides = set()
         obj.commands = set()
+        obj.conditions = VANILLA_CONDITIONS
```
Comment (WingedSeal) [inline src/jmc/compile/header.py:91]: This will permanently modify the original dict, you have to copy it.

**Review by WingedSeal — CHANGES_REQUESTED** [review]

File: src/jmc/compile/header.py (hunk 4 above)
Comment (Nico314159) [reply src/jmc/compile/header.py:91]: You mean set?

Comment (WingedSeal) [reply src/jmc/compile/header.py:91]: Yes, I meant set

**Comment by Nico314159** [comment]
Anything else I need to do?

**Pull request merged**